src/BlockTypes/ProductResultsCount.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+use Automattic\WooCommerce\Blocks\Utils\StyleAttributesUtils;
+
+/**
+ * ProductResultsCount class.
+ */
+class ProductResultsCount extends AbstractBlock {
+
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'product-results-count';
+
+	/**
+	 * Get the frontend script handle for this block type.
+	 *
+	 * @param string $key Data to get, or default to everything.
+	 */
+	protected function get_block_type_script( $key = null ) {
+		return null;
+	}
+
+	/**
+	 * Render the block.
+	 *
+	 * @param array    $attributes Block attributes.
+	 * @param string   $content Block content.
+	 * @param WP_Block $block Block instance.
+	 *
+	 * @return string Rendered block output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		ob_start();
+		woocommerce_result_count();
+		$product_results_count = ob_get_clean();
+
+		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
+		$classname          = isset( $attributes['className'] ) ? $attributes['className'] : '';
+
+		return sprintf(
+			'<div class="woocommerce wc-block-product-results-count %s %s">%s</div>',
+			esc_attr( $classes_and_styles['classes'] ),
+			$classname,
+			$product_results_count
+		);
```

Comment:
WordPress CS used to throw a warning/error about `Multiple placeholders should be ordered.` Not sure why it doesn't anymore. But I still think we should apply that practice. 

Also, look like the style is missing.

```suggestion
		return sprintf(
			'<div class="woocommerce wc-block-product-results-count %1$s %2$s" style="%3$s">%4$s</div>',
			esc_attr( $classes_and_styles['classes'] ),
			$classname,
			esc_attr( $classes_and_styles['styles'] ),
			$product_results_count
		);
```